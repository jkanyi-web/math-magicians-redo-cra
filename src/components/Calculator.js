import { useState } from 'react';
import calculate from '../logic/calculate';

export default function Calculator() {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const buttonClick = (e) => {
    setState(calculate(state, e.target.textContent));
  };

  return (
    <section className="container">
      <input
        type="text"
        className="input"
        value={state.next || state.total || '0'}
      />
      <div className="itemsContainer">
        <button type="button" onClick={buttonClick} className="items">
          AC
        </button>
        <button type="button" onClick={buttonClick} className="items">
          +/-
        </button>
        <button type="button" onClick={buttonClick} className="items">
          %
        </button>
        <button type="button" onClick={buttonClick} className="items orange">
          รท
        </button>
        <button type="button" onClick={buttonClick} className="items">
          7
        </button>
        <button type="button" onClick={buttonClick} className="items">
          8
        </button>
        <button type="button" onClick={buttonClick} className="items">
          9
        </button>
        <button type="button" onClick={buttonClick} className="items orange">
          x
        </button>
        <button type="button" onClick={buttonClick} className="items">
          4
        </button>
        <button type="button" onClick={buttonClick} className="items">
          5
        </button>
        <button type="button" onClick={buttonClick} className="items">
          6
        </button>
        <button type="button" onClick={buttonClick} className="items orange">
          -
        </button>
        <button type="button" onClick={buttonClick} className="items">
          1
        </button>
        <button type="button" onClick={buttonClick} className="items">
          2
        </button>
        <button type="button" onClick={buttonClick} className="items">
          3
        </button>
        <button type="button" onClick={buttonClick} className="items orange">
          +
        </button>
        <button type="button" onClick={buttonClick} className="items long">
          0
        </button>
        <button type="button" onClick={buttonClick} className="items">
          .
        </button>
        <button type="button" onClick={buttonClick} className="items orange">
          =
        </button>
      </div>
    </section>
  );
}
