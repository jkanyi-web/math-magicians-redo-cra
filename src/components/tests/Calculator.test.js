import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Calculator', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });

  it('should update the display and match the snapshot after clicking associated buttons', () => {
    const { getByText, container } = render(<Calculator />);
    const display = getByText('0');

    // Simulate button clicks
    fireEvent.click(getByText('7'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));

    // Check if the display is updated
    expect(display).toHaveTextContent('10');

    // Match the snapshot
    expect(container).toMatchSnapshot();
  });
});
