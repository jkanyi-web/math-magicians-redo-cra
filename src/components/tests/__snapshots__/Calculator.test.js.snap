// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <input
      class="input"
      readonly=""
      type="text"
      value="0"
    />
    <div
      class="itemsContainer"
    >
      <button
        class="items"
        type="button"
      >
        AC
      </button>
      <button
        class="items"
        type="button"
      >
        +/-
      </button>
      <button
        class="items"
        type="button"
      >
        %
      </button>
      <button
        class="items orange"
        type="button"
      >
        รท
      </button>
      <button
        class="items"
        type="button"
      >
        7
      </button>
      <button
        class="items"
        type="button"
      >
        8
      </button>
      <button
        class="items"
        type="button"
      >
        9
      </button>
      <button
        class="items orange"
        type="button"
      >
        x
      </button>
      <button
        class="items"
        type="button"
      >
        4
      </button>
      <button
        class="items"
        type="button"
      >
        5
      </button>
      <button
        class="items"
        type="button"
      >
        6
      </button>
      <button
        class="items orange"
        type="button"
      >
        -
      </button>
      <button
        class="items"
        type="button"
      >
        1
      </button>
      <button
        class="items"
        type="button"
      >
        2
      </button>
      <button
        class="items"
        type="button"
      >
        3
      </button>
      <button
        class="items orange"
        type="button"
      >
        +
      </button>
      <button
        class="items long"
        type="button"
      >
        0
      </button>
      <button
        class="items"
        type="button"
      >
        .
      </button>
      <button
        class="items orange"
        type="button"
      >
        =
      </button>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Calculator should update the display and match the snapshot after clicking associated buttons 1`] = `
<div>
  <section
    class="container"
  >
    <input
      class="input"
      type="text"
      value="10"
    />
    <div
      class="itemsContainer"
    >
      <button
        class="items"
        type="button"
      >
        AC
      </button>
      <button
        class="items"
        type="button"
      >
        +/-
      </button>
      <button
        class="items"
        type="button"
      >
        %
      </button>
      <button
        class="items orange"
        type="button"
      >
        รท
      </button>
      <button
        class="items"
        type="button"
      >
        7
      </button>
      <button
        class="items"
        type="button"
      >
        8
      </button>
      <button
        class="items"
        type="button"
      >
        9
      </button>
      <button
        class="items orange"
        type="button"
      >
        x
      </button>
      <button
        class="items"
        type="button"
      >
        4
      </button>
      <button
        class="items"
        type="button"
      >
        5
      </button>
      <button
        class="items"
        type="button"
      >
        6
      </button>
      <button
        class="items orange"
        type="button"
      >
        -
      </button>
      <button
        class="items"
        type="button"
      >
        1
      </button>
      <button
        class="items"
        type="button"
      >
        2
      </button>
      <button
        class="items"
        type="button"
      >
        3
      </button>
      <button
        class="items orange"
        type="button"
      >
        +
      </button>
      <button
        class="items long"
        type="button"
      >
        0
      </button>
      <button
        class="items"
        type="button"
      >
        .
      </button>
      <button
        class="items orange"
        type="button"
      >
        =
      </button>
    </div>
  </section>
</div>
`;
